jerry-core/ecma/operations/ecma-arraybuffer-object.c

```diff
@@ -36,6 +36,39 @@
  * @{
  */
 
+/**
+ * Helper function: check if the value is Array or SharedArray buffer
+ *
+ * @return true - if value is an ArrayBuffer or SharedArrayBuffer object
+ *         false - otherwise
+ */
+extern inline bool JERRY_ATTR_ALWAYS_INLINE
+ecma_is_array_or_sharedarray_buffer (const jerry_value_t value) /**< the target value */
+{
+#if JERRY_BUILTIN_SHAREDARRAYBUFFER
+  return (ecma_is_arraybuffer (value) || ecma_is_shared_arraybuffer (value));
+#else /* !JERRY_BUILTIN_SHAREDARRAYBUFFER */
+  return ecma_is_arraybuffer (value);
+#endif /* !JERRY_BUILTIN_SHAREDARRAYBUFFER */
```

Comment:
Afaik, this can be simplified (to reduce the #if-#else guards):
```c
  return ecma_is_arraybuffer (value)
#if JERRY_BUILTIN_SHAREDARRAYBUFFER
         || ecma_is_shared_arraybuffer (value)
#endif /* JERRY_BUILTIN_SHAREDARRAYBUFFER */
         || false;
```

There are similar cases for such formatting, check the jerry.c api file.

Reply:
Sidenote, the #endif guard's comment is incorrect, the original #if value should be present not the one used for the #else comment. (aka the comment should be: `/* JERRY_BUILTIN_SHAREDARRAYBUFFER */`). But if we adopt the code mentioned above this does not matter anymore.